lib/pbench/agent/tool_data_sink.py

```diff
@@ -1242,6 +1242,14 @@ def _valid_data(self, data):
             else:
                 return (action, directory, args)
 
+    @staticmethod
+    def check_failed_tool_init(tms: dict) -> bool:
```

Comment:
Seems like this method, `check_failed_tool_init` should be shared with what looks like the same code in `pbench-tool-meister-start`.

Reply:
I will check that :)

Reply:
I couldn't find a similar looking code in `pbench-tool-meister-start`. Are you referring to `record_tms` function in `tool_data_sink.py`?

Reply:
See https://github.com/distributed-system-analysis/pbench/pull/2659/files#diff-28fc2747e424d2014f9e3b3dd3e834315fa6db5a6079f62185988250ed197f69R453.